algonaut_transaction/src/account.rs

```diff
@@ -87,177 +84,131 @@ impl Account {
         })
     }
 
-    /// Sign a transaction with the account's private key
-    pub fn sign_transaction(
+    /// Sign transaction and generate a single signature SignedTransaction
+    pub fn sign_and_generate_signed_transaction(
+        &self,
+        transaction: &Transaction,
+    ) -> Result<SignedTransaction, AlgorandError> {
+        Ok(SignedTransaction {
+            transaction: transaction.clone(),
+            transaction_id: transaction.id()?,
```

Comment:
The little optimization with reusing the bytes to sign for the id generation and signing got lost here, but that seems negligible.